### PR TITLE
AGENT-204: Multi-arch support for OCP build

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
Ref: https://issues.redhat.com/browse/AGENT-204

### CODE REVIEW

**Comment by pawanpinjarkar** [comment]
Local testing result
Screenshot from 2022-05-18 21 03 31

File: Dockerfile.ocp [hunk 1]
```diff
@@ -21,8 +21,6 @@ COPY --from=builder /go/src/github.com/openshift/assisted-installer-agent/build/
 COPY --from=builder /go/src/github.com/openshift/assisted-installer-agent/build/domain_resolution /usr/bin/domain_resolution
 COPY --from=builder /go/src/github.com/openshift/assisted-installer-agent/scripts/installer/* /usr/local/bin/
 
-RUN INSTALL_PKGS="biosdevname dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools" && \
-    dnf install -y $INSTALL_PKGS && \
-    dnf clean all
+RUN if [[ "$(arch)" == "x86_64" ]]; then INSTALL_PKGS="biosdevname dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools" && dnf install -y $INSTALL_PKGS && dnf clean all; else INSTALL_PKGS="dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools" && dnf install -y $INSTALL_PKGS && dnf clean all; fi
```
Comment (osherdp) [inline Dockerfile.ocp:24]: just to make it easier on the eye:
```suggestion
RUN if [[ "$(arch)" == "x86_64" ]]; then \
    INSTALL_PKGS="dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools biosdevname"; else \
    INSTALL_PKGS="dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools"; fi \
    && dnf install -y $INSTALL_PKGS && dnf clean all
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: Dockerfile.ocp (hunk 1 above)
Comment (tsorya) [reply Dockerfile.ocp:24]: not nicer just to install install all packages that are the same in one line and special x86 one in another one?

Comment (osherdp) [reply Dockerfile.ocp:24]: Also an option, while not supporting providing two non-inclusive lists
Anything that is not just one long line of code

Comment (pawanpinjarkar) [reply Dockerfile.ocp]: Done

File: Dockerfile.ocp [hunk 2]
```diff
@@ -21,8 +21,8 @@ COPY --from=builder /go/src/github.com/openshift/assisted-installer-agent/build/
 COPY --from=builder /go/src/github.com/openshift/assisted-installer-agent/build/domain_resolution /usr/bin/domain_resolution
 COPY --from=builder /go/src/github.com/openshift/assisted-installer-agent/scripts/installer/* /usr/local/bin/
 
-RUN INSTALL_PKGS="biosdevname dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools" && \
-    dnf install -y $INSTALL_PKGS && \
-    dnf clean all
+RUN if [[ "$(arch)" == "x86_64" ]]; then INSTALL_PKGS="biosdevname" && dnf install -y $INSTALL_PKGS;fi
+RUN INSTALL_PKGS="dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools" && dnf install -y $INSTALL_PKGS
+RUN dnf clean all
```
Comment (osherdp) [inline Dockerfile.ocp:26]: nit: ``INSTALL_PKGS`` in current implementation is redundant
instead:
```suggestion
RUN if [[ "$(arch)" == "x86_64" ]]; then dnf install -y biosdevname && dnf clean all; fi
RUN dnf install -y dhclient dmidecode file findutils fio ipmitool iputils nmap openssh-clients podman chrony smartmontools" && dnf clean all
```

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by celebdor** [comment]
I agree that the variable is not useful in the way it is done. It doesn't hurt either...
/lgtm

**Comment by lranjbar** [comment]
I agree with the nit. I think this is alright for now since we have a larger problem it seems in assisted-service.

/approve

**Comment by osherdp** [comment]
/hold
```
/bin/sh: -c: line 0: unexpected EOF while looking for matching `"'
/bin/sh: -c: line 1: syntax error: unexpected end of file
```

**Review by osherdp — APPROVED** [review]

**Pull request merged**